Aztec/Classes/Converter/CLinkedListToArrayConverter.swift

```diff
@@ -101,21 +101,20 @@ class SafeCLinkedListToArrayConverter<ElementConverterType: SafeConverter>: Safe
     ///                     linked list into an element of the output array.
     ///     - next: the closure that will provide the next element in the list, given the current one.
     ///
-    init(elementConverter: ElementConverterType, next: (ElementConverterType.TypeIn) -> (TypeIn)) {
+    init(elementConverter: ElementConverterType, next: @escaping (ElementConverterType.TypeIn) -> (TypeIn)) {
         self.elementConverter = elementConverter
         self.next = next
     }
 
-    func convert(inputPtr: TypeIn) -> TypeOut {
+    func convert(_ inputPtr: TypeIn) -> TypeOut {
         var result = TypeOut()
-        var currentPtr = inputPtr
-
-        while currentPtr != nil {
-            let element = currentPtr.memory
+        var nextPtr: UnsafeMutablePointer<ElementConverterType.TypeIn>? = inputPtr
```

Comment:
Same as above. TypeIn is already an `UnsafeMutablePointer`, do we need to re-wrap-it?